### ISSUE TITLE
Something caused UMS to shutdown

### ISSUE THREAD

**Issue by Mik-S-UMS** [UniversalMediaServer/UniversalMediaServer#3775]
UMS just suddenly shutdown for no reason. I only noticed it as I have the media player for the PS4 on while I was doing other things and saw UMS disappear from it.

I was going over [stuff](https://github.com/UniversalMediaServer/UniversalMediaServer/discussions/3773) earlier experimenting with the web player and I remember it doing it then too but just thought it was a fluke.

I don't know exactly what I did so don't know how to replicate, but I think I started to play one video and when it did not start immediately I went to another page. It was not until about an hour later that UMS shutdown though.

There are a few errors in the logs and I think this one about a key violation in thumbnails belonged to the video I played.
```
ERROR 2023-01-13 01:05:19.011 [File event] Database "medias" error while writing md5 "a2f5c106146668456a4ac70ee6e24b83" in "THUMBNAILS" for "D:\Users\Mik\Downloads\Video\Watched\Dont Trust The B -   In Apartment 23 S01E03.mkv": Unique index or primary key violation: "PUBLIC.MD5_IDX ON PUBLIC.THUMBNAILS(MD5 NULLS FIRST) VALUES ( /* 13901 */ 'a2f5c106146668456a4ac70ee6e24b83' )"; SQL statement:
INSERT INTO THUMBNAILS (THUMBNAIL, MODIFIED, MD5) VALUES (?, ?, ?) [23505-214]
TRACE 2023-01-13 01:05:19.011 [File event] 
org.h2.jdbc.JdbcSQLIntegrityConstraintViolationException: Unique index or primary key violation: "PUBLIC.MD5_IDX ON PUBLIC.THUMBNAILS(MD5 NULLS FIRST) VALUES ( /* 13901 */ 'a2f5c106146668456a4ac70ee6e24b83' )"; SQL statement:
INSERT INTO THUMBNAILS (THUMBNAIL, MODIFIED, MD5) VALUES (?, ?, ?) [23505-214]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:508)
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:477)
	at org.h2.message.DbException.get(DbException.java:223)
	at org.h2.message.DbException.get(DbException.java:199)
	at org.h2.index.Index.getDuplicateKeyException(Index.java:525)
	at org.h2.mvstore.db.MVSecondaryIndex.checkUnique(MVSecondaryIndex.java:223)
	at org.h2.mvstore.db.MVSecondaryIndex.add(MVSecondaryIndex.java:184)
	at org.h2.mvstore.db.MVTable.addRow(MVTable.java:519)
	at org.h2.command.dml.Insert.insertRows(Insert.java:174)
	at org.h2.command.dml.Insert.update(Insert.java:135)
	at org.h2.command.CommandContainer.executeUpdateWithGeneratedKeys(CommandContainer.java:242)
	at org.h2.command.CommandContainer.update(CommandContainer.java:163)
	at org.h2.command.Command.executeUpdate(Command.java:252)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdateInternal(JdbcPreparedStatement.java:209)
	at org.h2.jdbc.JdbcPreparedStatement.executeUpdate(JdbcPreparedStatement.java:169)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
	at net.pms.database.MediaTableThumbnails.setThumbnail(MediaTableThumbnails.java:218)
	at net.pms.database.MediaTableThumbnails.setThumbnail(MediaTableThumbnails.java:153)
	at net.pms.dlna.DLNAResource.checkThumbnail(DLNAResource.java:3432)
	at net.pms.dlna.RealFile.checkThumbnail(RealFile.java:383)
	at net.pms.dlna.RealFile.isValid(RealFile.java:135)
	at net.pms.dlna.RootFolder.parseFileForDatabase(RootFolder.java:1308)
	at net.pms.dlna.RootFolder.lambda$static$4(RootFolder.java:1269)
	at net.pms.util.FileWatcher$Notice.run(FileWatcher.java:433)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```
after that there are several of this error
```
ERROR 2023-01-13 02:58:27.470 [pool-6-thread-2] RuntimeException in PlayerApiServlet: Cannot invoke "java.net.InetAddress.getHostAddress()" because the return value of "net.pms.renderers.devices.WebGuiRenderer.getAddress()" is null
TRACE 2023-01-13 02:58:27.470 [pool-6-thread-2] {}
java.lang.NullPointerException: Cannot invoke "java.net.InetAddress.getHostAddress()" because the return value of "net.pms.renderers.devices.WebGuiRenderer.getAddress()" is null
	at net.pms.renderers.devices.WebGuiRenderer.start(WebGuiRenderer.java:242)
	at net.pms.network.webguiserver.servlets.PlayerApiServlet.sendMedia(PlayerApiServlet.java:1038)
	at net.pms.network.webguiserver.servlets.PlayerApiServlet.doGet(PlayerApiServlet.java:186)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:497)
	at net.pms.network.webguiserver.GuiHttpServlet.service(GuiHttpServlet.java:42)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:584)
	at net.pms.network.httpserverservletcontainer.HttpHandlerServlet.handle(HttpHandlerServlet.java:37)
	at jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Unknown Source)
	at jdk.httpserver/sun.net.httpserver.AuthFilter.doFilter(Unknown Source)
	at jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Unknown Source)
	at jdk.httpserver/sun.net.httpserver.ServerImpl$Exchange$LinkHandler.handle(Unknown Source)
	at jdk.httpserver/com.sun.net.httpserver.Filter$Chain.doFilter(Unknown Source)
	at jdk.httpserver/sun.net.httpserver.ServerImpl$Exchange.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

It appears to be running out of memory for some reason, possibly due to ffmpeg trying to stream to nothing
```
INFO  2023-01-13 02:58:28.819 [ffmpeg.exe-95] FATAL ERROR: OutOfMemory / dumping stats
TRACE 2023-01-13 02:58:28.822 [UMS Shutdown] Stopping the network scanner
```
As I can't get a clean log to replicate this I have attached the whole log before restarting UMS which may be a lot to dig though. Luckily I had put UMS in trace mode earlier.
[debug.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/10408122/debug.zip)

There was another error relating to musicbrainz.org which is probably not related but worth investigating.
```
ERROR 2023-01-13 01:03:13.313 [Library Scanner] Could not lookup audio cover for "": musicbrainz.org replied with status code 400
```

**Comment by SubJunk** [comment]
Thanks for this, looks like some useful errors here. I'll have a look soon

**Comment by SubJunk** [comment]
@Mik-S-UMS has this happened again? I'm not sure it's a problem anymore

**Comment by Mik-S-UMS** [comment]
Not seen this happen again but I have not been messing around like I was then so I have no idea what caused it.

I have been meaning to revisit that discussion to update it with the current version but have not had much time recently. When I do I'll see if I can trigger it again.

I think some of the errors were due to poor/missing tags in some of my mp3s. Some I had no idea what they were and took a while to identify them as there were part of a compilation. I think I have fixed most of them now.
